### PR TITLE
Add release process, clear out releasenotes.md

### PR DESCRIPTION
**Pull request by jviau**

### CODE REVIEW

File: doc/release_process.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    - We follow an approach of just releasing everything, even if the package has no changes. _Unless_ we intentionally do not want to release a package.
+2. Ensure appropriate `RELEASENOTES.md` are updated in the repository.
+    - These are per-package and contain only that packages changes.
+    - The contents will be included in the `.nupkg` and show up in nuget.org's release notes tab.
+    - Clear these files after a release.
+3. Ensure `CHANGELOG.md` in repo root is updated.
+4. Tag the release.
+    - `git tag v<version>`, `git push <remote> -u <tag>`
+5. Draft github release for new tag
+6. Kick off ADO release build (use the tag as the build target, enter `refs/tags/<tag>`)
```
Comment (bachuv) [inline doc/release_process.md:15]: Can you add a link to the pipeline?

File: doc/release_process.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+7. Validate signing, package contents (if build changes were made)
+8. Create ADO release
```
Comment (bachuv) [inline doc/release_process.md:17]: Can you add a link and more details about this step?

File: doc/release_process.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+9. Release to ADO feed
+10. Internal validation
```
Comment (bachuv) [inline doc/release_process.md:19]: Does this step just mean a smoke test is there anything else we should validate?

Comment (jviau) [reply doc/release_process.md:19]: I'll just remove this, don't have concrete steps.

File: doc/release_process.md (hunk 2 above)
Comment (jviau) [reply doc/release_process.md:17]: Added clarification on how to create the release from a build.

File: doc/release_process.md (hunk 1 above)
Comment (jviau) [reply doc/release_process.md:15]: Added

**Review by bachuv — APPROVED** [review]

**Pull request merged**